client/state/selectors/is-site-coming-soon.js

```diff
@@ -27,8 +22,11 @@ export default function isSiteComingSoon( state, siteId ) {
 	const settings = getSiteSettings( state, siteId );
 
 	if ( settings ) {
-		// Site settings returns a numerical value for wpcom_coming_soon.
-		return parseInt( settings.wpcom_coming_soon, 10 ) === 1;
+		// Site settings returns a numerical value for wpcom_public_coming_soon.
+		return (
+			parseInt( settings.wpcom_coming_soon, 10 ) === 1 ||
+			parseInt( settings.wpcom_public_coming_soon, 10 ) === 1
+		);
```

Comment:
This is a tough choice. Having the `isSiteComingSoon` selector return true for either v1 or v2 makes sense because it means it'll often just show the sort of copy in the UI that we want (the feature is still named "Coming Soon" after all).

But `isSiteComingSoon` is also used for showing notices about why SEO and podcasting settings don't work. And those are only applicable for v1. Ultimately those SEO and podcasting messages never show because of other booleans `if`ing them out. But using `isSiteComingSoon` for both v1 and v2 sort of makes the logic even more confusing than it currently is.

On the other hand, it'd be sort of gross to have a `_v1` and `_v2` version of the same selector.

Sorry, I don't really know which way I'm advocating tbh.

Reply:
I'll take a look at where this is used, I'm more than ok with a v2 suffix but those particular examples look like work arounds for v1 shortcomings rather than coming soon feature specific things.

In that case it is probably better to v2 flag check on those work arounds individually.

Reply:
Duplicated this function into a V2 version and audited the usages of V1, added v2 where it made sense. Could use some eyes on those changes - they're all untested and I'm not really sure how to get to some of the states I've touched in this PR now.